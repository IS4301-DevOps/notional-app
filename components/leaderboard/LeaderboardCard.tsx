import React from 'react';
import Image from 'next/image';
import Link from 'next/link';

export default function LeaderboardCard(props) {
  return (
    <Link href='/leaderboard/badges-collection'>
      <div
        className={`relative flex items-center space-x-3 rounded-lg border border-gray-300 drop-shadow-md ${
          props.index === 0 ? 'bg-[#F7F6DC]' : 'bg-white'
        } px-6 py-5 shadow-sm focus-within:ring-2 focus-within:ring-indigo-500 focus-within:ring-offset-2 hover:border-gray-400`}
      >
        <div className='flex-shrink-0'>
          <img className='h-[75px] w-[75px] rounded-full' src={props.person.imageUrl} alt='' />
          {/* <Image alt='me' width='64' height='64' src='/badges/sparkly_trees_25.svg' /> */}
        </div>
        <div className='min-w-0 flex-1'>
          <a href='#' className='focus:outline-none'>
            <span className='absolute inset-0' aria-hidden='true' />
            <p className='text-sm font-bold text-gray-900 mb-3'>{props.person.name}</p>
            {props.person.badges.map((badge, index) => (
              <span key={index} className='mr-3'>
                <Image key={index} alt='me' width='45' height='45' src={`/badges/${badge}.svg`} />
              </span>
            ))}
            {/* <p className='truncate text-sm text-gray-500'>{props.person.role}</p> */}
          </a>
        </div>

        <div className='translate-x-6'>
          <span
            style={{ backgroundColor: `${props.person.color}` }}
            className={`ml-10 mb-2 inline-flex h-7 w-7 items-center justify-center rounded-full drop-shadow-lg`}
          >
            <span className='font-bold text-xs leading-none text-black'>{props.person.ranking}</span>
          </span>
          <div>
            <span className='relative z-10'>
              <Image alt='me' width='35' height='35' src='/badges/leaderboardtree.png' />
            </span>
            <span className='font-bold relative inline-flex items-center -translate-x-4 -translate-y-3 rounded-md bg-gray-100 px-2.5 py-0.5 text-sm text-gray-800'>
              {props.person.treesPlanted}
            </span>
          </div>
        </div>
      </div>
    </Link>
  );
}
