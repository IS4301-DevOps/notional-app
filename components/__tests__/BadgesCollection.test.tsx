import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import LeaderboardCard from '../leaderboard/LeaderboardCard';

afterEach(() => {
  cleanup();
});

test('should render LeaderboardCard component', () => {
  const person = {
    name: 'David',
    badges: ['carbon_100', 'climate_positive_6'],
    imageUrl:
      'https://images.unsplash.com/profile-1602788303253-d8e1b652c4c9image?dpr=1&auto=format&fit=facearea&facepad=2.25&w=256&h=256&q=80',
    ranking: '4',
    treesPlanted: '1132',
    color: '#FFFF',
  };
  const { getByText } = render(<LeaderboardCard person={person} />);
  getByText('David');
});

test('matches snapshot', () => {
  const person = {
    name: 'David',
    badges: ['carbon_100', 'climate_positive_6'],
    imageUrl:
      'https://images.unsplash.com/profile-1602788303253-d8e1b652c4c9image?dpr=1&auto=format&fit=facearea&facepad=2.25&w=256&h=256&q=80',
    ranking: '4',
    treesPlanted: '1132',
    color: '#FFFF',
  };
  const tree = renderer.create(<LeaderboardCard person={person} />).toJSON();
  expect(tree).toMatchSnapshot();
});
