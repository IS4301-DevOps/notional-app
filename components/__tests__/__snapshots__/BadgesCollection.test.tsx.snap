// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="relative flex items-center space-x-3 rounded-lg border border-gray-300 drop-shadow-md bg-white px-6 py-5 shadow-sm focus-within:ring-2 focus-within:ring-indigo-500 focus-within:ring-offset-2 hover:border-gray-400"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="flex-shrink-0"
  >
    <img
      alt=""
      className="h-[75px] w-[75px] rounded-full"
      src="https://images.unsplash.com/profile-1602788303253-d8e1b652c4c9image?dpr=1&auto=format&fit=facearea&facepad=2.25&w=256&h=256&q=80"
    />
  </div>
  <div
    className="min-w-0 flex-1"
  >
    <a
      className="focus:outline-none"
      href="#"
    >
      <span
        aria-hidden="true"
        className="absolute inset-0"
      />
      <p
        className="text-sm font-bold text-gray-900 mb-3"
      >
        David
      </p>
      <span
        className="mr-3"
      >
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2745%27%20height=%2745%27/%3e"
              style={
                {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="me"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="me"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/badges/carbon_100.svg"
              srcSet="/badges/carbon_100.svg 1x, /badges/carbon_100.svg 2x"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </span>
      <span
        className="mr-3"
      >
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2745%27%20height=%2745%27/%3e"
              style={
                {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="me"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="me"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/badges/climate_positive_6.svg"
              srcSet="/badges/climate_positive_6.svg 1x, /badges/climate_positive_6.svg 2x"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </span>
    </a>
  </div>
  <div
    className="translate-x-6"
  >
    <span
      className="ml-10 mb-2 inline-flex h-7 w-7 items-center justify-center rounded-full drop-shadow-lg"
      style={
        {
          "backgroundColor": "#FFFF",
        }
      }
    >
      <span
        className="font-bold text-xs leading-none text-black"
      >
        4
      </span>
    </span>
    <div>
      <span
        className="relative z-10"
      >
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2735%27%20height=%2735%27/%3e"
              style={
                {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="me"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="me"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fbadges%2Fleaderboardtree.png&w=96&q=75"
              srcSet="/_next/image?url=%2Fbadges%2Fleaderboardtree.png&w=48&q=75 1x, /_next/image?url=%2Fbadges%2Fleaderboardtree.png&w=96&q=75 2x"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </span>
      <span
        className="font-bold relative inline-flex items-center -translate-x-4 -translate-y-3 rounded-md bg-gray-100 px-2.5 py-0.5 text-sm text-gray-800"
      >
        1132
      </span>
    </div>
  </div>
</div>
`;
